apps/launch/hooks/useGetMyStakingContracts.ts

```diff
@@ -0,0 +1,175 @@
+import { kebabCase } from 'lodash';
+import { useEffect, useState } from 'react';
+import { Abi, Address, parseAbiItem } from 'viem';
+import { mainnet } from 'viem/chains';
+import { useAccount, usePublicClient, useReadContract, useReadContracts } from 'wagmi';
+
+import { GATEWAY_URL, HASH_PREFIX } from 'libs/util-constants/src';
+import {
+  STAKING_FACTORY,
+  STAKING_TOKEN,
+  VOTE_WEIGHTING,
+} from 'libs/util-contracts/src/lib/abiAndAddresses';
+import { getBytes32FromAddress } from 'libs/util-functions/src';
+
+import { SupportedChain } from 'common-util/constants/rpcs';
+import { CONTRACT_TEMPLATES } from 'common-util/constants/stakingContract';
+import { useAppDispatch } from 'store/index';
+import { setMyStakingContracts } from 'store/launch';
+import { MyStakingContract } from 'types/index';
+
+type Metadata = { name: string; description: string };
+
+const getMetadata = async (tokenUri: undefined | null | string) => {
+  if (!tokenUri) return null;
+
+  try {
+    const uri = `${HASH_PREFIX}${tokenUri.substring(2)}`;
+    const ipfsUrl = `${GATEWAY_URL}${uri}`;
+    const response = await fetch(ipfsUrl);
+    const json = await response.json();
+    return json;
+  } catch (e) {
+    window.console.error(e);
+  }
+
+  return null;
+};
+
+const useGetAllNominees = () => {
+  // it is only available on mainnet, hence hardcoded.
+  const { data: nominees } = useReadContract({
+    address: (VOTE_WEIGHTING.addresses as Record<number, Address>)[mainnet.id],
+    abi: VOTE_WEIGHTING.abi,
+    chainId: mainnet.id,
+    functionName: 'getAllNominees',
+    args: [],
+    query: {
+      select: (data: unknown) => {
+        const list = data as { account: Address; chainId: bigint }[];
+        return list.filter((nominee) => nominee.chainId === BigInt(mainnet.id));
```

Comment:
I don't think we need to filter only those on mainnet, in this case we won't see any nominated contracts e.g. on gnosis. Vote Weighting contracts live only on mainnet, but contains all nominees from different chains with chainId specified, so we can check if any contracts on different chains are nominated